File: src/main/java/org/checkerframework/checker/mungo/typestate/graph/states/DecisionState.java
package org.checkerframework.checker.mungo.typestate.graph.states;

import org.checkerframework.checker.mungo.typestate.ast.TDecisionNode;
import org.checkerframework.checker.mungo.typestate.ast.TDecisionStateNode;
import org.checkerframework.checker.mungo.typestate.graph.states.AbstractState;

public class DecisionState extends AbstractState<TDecisionStateNode, TDecisionNode> {

  public DecisionState(TDecisionStateNode node) {
    super(node);
  }

  @Override
  public String toString() {
    return "DecisionState{node=" + node + "}";
  }
}

File: src/main/kotlin/org/checkerframework/checker/mungo/typestate/graph/states/DecisionState.kt
package org.checkerframework.checker.mungo.typestate.graph.states

import org.checkerframework.checker.mungo.typestate.ast.TDecisionNode
import org.checkerframework.checker.mungo.typestate.ast.TDecisionStateNode
import java.util.HashMap

class DecisionState(node: TDecisionStateNode?) : AbstractState<TDecisionStateNode>(node) {

  val transitions: MutableMap<TDecisionNode, State> = HashMap()

  fun addTransition(transition: TDecisionNode, destination: State) {
    transitions[transition] = destination
  }

  override fun toString(): String {
    return "DecisionState{node=$node}"
  }
}
